pkg/genevalogging/genevalogging.go

```diff
@@ -0,0 +1,478 @@
+			<IdentityComponent name="Region">{{ .Region | Escape }}</IdentityComponent>
+			<IdentityComponent name="SubscriptionId">{{ .SubscriptionId | Escape }}</IdentityComponent>
+			<IdentityComponent name="ResourceGroupName">{{ .ResourceGroupName | Escape }}</IdentityComponent>
+			<IdentityComponent name="ResourceName">{{ .ResourceName | Escape }}</IdentityComponent>
+		</Identity>
+		<AgentResourceUsage diskQuotaInMB="50000"/>
+	</Management>
+	<Sources>
+		<Source name="journald" dynamic_schema="true"/>
+		<Source name="audit" dynamic_schema="true"/>
+	</Sources>
+	<Events>
+		<MdsdEvents>
+			<MdsdEventSource source="journald">
+				<RouteEvent eventName="CustomerSyslogEvents" storeType="CentralBond" priority="High"/>
+			</MdsdEventSource>
+			<MdsdEventSource source="audit">
+				<RouteEvent eventName="CustomerAuditLogEvents" storeType="CentralBond" priority="High"/>
+			</MdsdEventSource>
+		</MdsdEvents>
+	</Events>
+</MonitoringManagement>`
+)
+
+var (
+	mdsdTemplate = template.Must(template.New("").Funcs(map[string]interface{}{
+		"Escape": func(s string) (string, error) {
+			var b bytes.Buffer
+			err := xml.EscapeText(&b, []byte(s))
+			return b.String(), err
+		},
+	}).Parse(mdsdTemplateStr))
+)
+
+type GenevaLogging interface {
+	CreateOrUpdate(ctx context.Context) error
+}
+
+type GenevaLoggingConfig struct {
+	Sector                  string
+	Namespace               string
+	Account                 string
+	ControlPlaneAccount     string
+	ControlPlaneEnvironment string
+	ControlPlaneRegion      string
+
+	Key  *rsa.PrivateKey
+	Cert *x509.Certificate
+
+	MDSDImage    string
+	TDAgentImage string
+}
+
+type genevaLogging struct {
+	location       string
+	subscriptionID string
+	clusterName    string
+	resourceGroup  string
+
+	config *GenevaLoggingConfig
+
+	kubeCli kubernetes.Interface
+}
+
+func New(location, subscriptionID, clusterName, resouceGroup string, config *GenevaLoggingConfig, cli kubernetes.Interface) GenevaLogging {
+	return &genevaLogging{
+		location:       location,
+		subscriptionID: subscriptionID,
+		clusterName:    clusterName,
+		resourceGroup:  resouceGroup,
+		config:         config,
+		kubeCli:        cli,
+	}
+}
+
+func (g *genevaLogging) mdsdConfig() (string, error) {
+	b := &bytes.Buffer{}
+	err := mdsdTemplate.Execute(b, map[string]string{
+		"Namespace":         g.config.Namespace,
+		"Account":           g.config.Account,
+		"Region":            g.location,
+		"SubscriptionId":    g.subscriptionID,
+		"ResourceName":      g.clusterName,
+		"ResourceGroupName": g.resourceGroup,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	return string(b.Bytes()), nil
+}
+
+func (g *genevaLogging) ensureNamespace() error {
+	exists := false
+	ready := false
+	for !exists || !ready {
+		ns, err := g.kubeCli.CoreV1().Namespaces().Get(namespace, metav1.GetOptions{})
+		if err != nil && !errors.IsNotFound(err) {
+			return err
+		}
+		if ns != nil {
+			exists = true
+			if len(ns.GetAnnotations()) > 0 {
```

Comment:
I am not sure without testing what to look for, so for now just look for some annotations. This should really be logged too..